bentoml/deployment/store.py

```diff
@@ -16,45 +16,92 @@
 from __future__ import division
 from __future__ import print_function
 
+import logging
+
 from sqlalchemy import Column, String, JSON
-from google.protobuf.json_format import MessageToDict
+from sqlalchemy.orm.exc import NoResultFound, MultipleResultsFound
+from google.protobuf.json_format import MessageToDict, ParseDict
 
+from bentoml.config import config
 from bentoml.db import Base, create_session
+from bentoml.proto import deployment_pb2
+
+
+logger = logging.getLogger(__name__)
 
 
 class Deployment(Base):
     __tablename__ = 'deployments'
 
     id = Column(String, primary_key=True)
     name = Column(String, unique=True)
-    namespace = Column(String)
+    namespace = Column(String, nullable=False)
 
     spec = Column(JSON)
-    labels = Column(JSON)
-    annotation = Column(JSON)
+    labels = Column(JSON, nullable=False, default={})
+    annotation = Column(JSON, nullable=False, default={})
+
+
+def deployment_pb_to_orm_obj(deployment_pb):
+    return Deployment(
+        name=deployment_pb.name,
+        namespace=deployment_pb.namespace,
+        spec=MessageToDict(deployment_pb.spec),
+        labels=MessageToDict(deployment_pb.labels),
+        annotation=MessageToDict(deployment_pb.labels),
+    )
+
+
+def deployment_orm_obj_to_pb(deployment_obj):
+    return deployment_pb2.Deployment(
+        name=deployment_obj.name,
+        namespace=deployment_obj.namespace,
+        spec=ParseDict(deployment_obj.spec, deployment_pb2.DeploymentSpec),
```

Comment:
I test this in ipython, and got 

```
AttributeError: 'google.protobuf.pyext._message.FieldProperty' object has no attribute 'SetInParent'
```
I got this error, because inside DeploymentSpec, we have additional messages like `SageMakerOperatorConfig` and others. We probably want to make sure parse those config with `ParseDict` first

Reply:
oh thank for catching this, that was a typo, should be ```ParseDict(deployment_obj.spec, deployment_pb2.DeploymentSpec())```